controllers/activegate/reconciler/capability/reconciler.go

```diff
@@ -98,6 +108,11 @@ func (r *Reconciler) Reconcile() (update bool, err error) {
 		if update || err != nil {
 			return update, errors.WithStack(err)
 		}
+
+		update, err = r.updateServiceIfOutdated()
```

Comment:
Just checking the annotation on the Dynakube should be enough. Then you can check if the Service on the cluster has the required fields or needs to be updated.

Reply:
I switched to comparing ports instead of setting internal annotations - works fine as well, though not sure about maintainability. Maybe not a problem though.